### ISSUE TITLE
Composite: `shiftFocus` broken in 0.4.12

### ISSUE THREAD

**Issue by ciampo** [ariakit/ariakit#4180]
### Current behavior

When upgrading from `0.4.11` to `0.4.12`, setting `shiftFocus={false}` for the `Composite` doesn't seem to have effect.

Found while working on https://github.com/WordPress/gutenberg/pull/65907

### Steps to reproduce the bug

1. Open https://stackblitz.com/edit/thqfra-ff5mtm?file=button%2Findex.tsx
2. Navigate to item B3
3. Press arrow up

### Expected behavior

The active item should continue to be B3, but instead it becomes A2, as if the `shiftFocus` option was set to `true`.

Note that the demo behaves as expected with`@ariakit/core` set to version `0.4.11`.

### Workaround

N/A

### Possible solutions

N/A

**Comment by diegohaz** [comment]
Thanks for reporting! It's probably related to #4164

**Comment by benrodrs** [comment]
Good catch! For simple cases, you can work around the `focusShift: false` issue by rendering a dummy disabled `CompositeItem` to fill the gaps in rows with fewer columns:

```jsx
<Ariakit.CompositeRow>
  <Ariakit.CompositeItem>Item A1</Ariakit.CompositeItem>
  <Ariakit.CompositeItem>Item A2</Ariakit.CompositeItem>
  <Ariakit.CompositeItem disabled hidden />
</Ariakit.CompositeRow>
```

I understand this isn't always possible, especially when you don't control the `CompositeItem` elements being rendered. There might be a way to automatically add this item to the composite state, but it would likely require too much work.

**Comment by ciampo** [comment]
> I understand this isn't always possible, especially when you don't control the CompositeItem elements being rendered.

It's not possible for our case, so I assume we may have to wait for the next release including the fix
